Classes/MapEditor/MapEditor.cs

```diff
@@ -579,4 +575,33 @@ public void Dispose()
             b.beat *= scaleFactor;
         }
     }
+    internal void ToggleSelectionForBookmark(Bookmark b)
+    {
+        if (currentMapDifficulty == null)
+        {
+            return;
+        }
+        double endBeat = currentMapDifficulty.bookmarks
+            .GetViewBetween(b, new Bookmark(double.PositiveInfinity, "songEnd"))
+            .Skip(1)
+            .Select(x => x.beat)
+            .FirstOrDefault(double.PositiveInfinity);
+
+        ToggleSelection(currentMapDifficulty.GetNotesRange(b.beat, endBeat));
```

Comment:
After testing more, it's very unintuitive that the old selection is not deselected and the notes get toggled instead of just being selected. This is not in line with the current behavior user gets when selecting notes by dragging or clicking them.

I often first do the notes just for the rhythm and then do the patterns in unique sections, then copy the patterns to a similar section and adjust if needed. If I go for the following sequence of inputs which are very natural with this approach, I end up deleting a section of the map that's not currently visible on screen:
1. Ctrl+Click on Bookmark 1 to select all notes
2. Ctrl+C to copy them to clipboard
3. Scroll to Bookmark 2
4. Ctrl+Click on Bookmark 2 to select all notes
5. Delete to remove old notes from Bookmark 2 (this accidently deletes notes from Bookmark 1 as well)
6. Ctrl+V to paste notes from Bookmark 1 to Bookmark 2.

This behavior of old selection being preserved and notes toggling should only occur when Shift is pressed at the same time as Ctrl when clicking on the bookmark/BPM change.